core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java

```diff
@@ -108,7 +111,26 @@ public List<String> getMetadataDatasourceIds() {
   }
 
   public MetadataTempFilesListDto uploadMetadataFilesToTempDir( InputStream metadataFile,
-      List<InputStream> localeFileStreams, List<String> localeFileNames ) throws Exception {
+     FormDataContentDisposition schemaFileInfo, List<InputStream> localeFileStreams, List<String> localeFileNames ) throws Exception {
+
+    String fileNameMain = schemaFileInfo.getFileName();
+    ZipInputStream zis = null;
```

Comment:
No added unit test for this section of new logic of extracting the "xmi" file from zip file.